src/test/java/uk/gov/pay/directdebit/payments/services/sandbox/SandboxPaymentStateCalculatorTest.java

```diff
@@ -16,54 +23,123 @@
 import uk.gov.pay.directdebit.payments.model.SandboxPaymentId;
 import uk.gov.pay.directdebit.webhook.sandbox.resources.WebhookSandboxResource.SandboxEventAction;
 
+import java.time.ZonedDateTime;
 import java.util.Optional;
 
+import static java.time.ZoneOffset.UTC;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 import static org.mockito.BDDMockito.given;
+import static uk.gov.pay.directdebit.events.model.GovUkPayEventType.PAYMENT_SUBMITTED;
 import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.SANDBOX;
 import static uk.gov.pay.directdebit.mandate.fixtures.MandateFixture.aMandateFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture.aGatewayAccountFixture;
+import static uk.gov.pay.directdebit.payments.fixtures.GovUkPayEventFixture.aGovUkPayEventFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.PaymentFixture.aPaymentFixture;
-import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxPaymentStateCalculator.SANDBOX_ACTIONS_THAT_CHANGE_STATE;
+import static uk.gov.pay.directdebit.payments.services.GovUkPayEventToPaymentStateMapper.GOV_UK_PAY_EVENT_TYPES_THAT_CHANGE_PAYMENT_STATE;
+import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxEventToPaymentStateMapper.SANDBOX_ACTIONS_THAT_CHANGE_PAYMENT_STATE;
 
-@RunWith(MockitoJUnitRunner.class)
+@RunWith(JUnitParamsRunner.class)
 public class SandboxPaymentStateCalculatorTest {
 
     private static final SandboxPaymentId SANDBOX_PAYMENT_ID = SandboxPaymentId.valueOf("Sandbox payment ID");
 
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
     @Mock
     private SandboxEvent mockSandboxEvent;
 
     @Mock
     private SandboxEventDao mockSandboxEventDao;
 
+    @Mock
+    private GovUkPayEventDao mockGovUkPayEventDao;
+
     @InjectMocks
     private SandboxPaymentStateCalculator sandboxPaymentStateCalculator;
-    
+
     private Payment payment;
-    
+
     @Before
     public void setUp() {
         GatewayAccountFixture gatewayAccountFixture = aGatewayAccountFixture()
                 .withPaymentProvider(SANDBOX);
 
         MandateFixture mandateFixture = aMandateFixture()
                 .withGatewayAccountFixture(gatewayAccountFixture);
-        
+
         payment = aPaymentFixture()
                 .withMandateFixture(mandateFixture)
                 .withPaymentProviderId(SANDBOX_PAYMENT_ID)
                 .toEntity();
+    }
+
+    @Test
+    @Parameters({
```

Comment:
Wondering if its necessary to use `@Parameters` when there's only one scenario, is this to make it simpler if future states are added?

Reply:
Yeh I was thinking it will make it a tiny bit easier in the future. I know that we're going to be adding another GovUkPayEvent. We don't have any plans to add another SandboxEvent, so maybe this isn't worthwhile but maybe it is.

Reply:
I don't feel strongly, the test is readable as-is. I like the parameterised tests for multiple scenarios as it removes quit a bit of boiler plate.